pkg/util/orchestrator/fake/client.go

```diff
@@ -47,22 +47,30 @@ func (o *FakeOrc) AddInstance(cluster, host string, master bool, sls int64, slav
 		ReadOnly: !master,
 		SlaveLagSeconds: NullInt64{
 			Valid: valid,
-			Int64: sls,
+			Int64: lag,
 		},
 		ClusterName:       cluster,
-		Slave_SQL_Running: slaveR,
-		Slave_IO_Running:  slaveR,
+		Slave_SQL_Running: slaveRunning,
+		Slave_IO_Running:  slaveRunning,
 		IsUpToDate:        upToDate,
 		IsLastCheckValid:  upToDate,
 	}
 	if o.Clusters == nil {
-		o.Clusters = make(map[string][]Instance)
+		o.Clusters = make(map[string][]*Instance)
 	}
 	clusters, ok := o.Clusters[cluster]
 	if ok {
-		o.Clusters[cluster] = append(clusters, inst)
+		o.Clusters[cluster] = append(clusters, &inst)
+	} else {
+		o.Clusters[cluster] = []*Instance{&inst}
 	}
-	o.Clusters[cluster] = []Instance{inst}
+
+	for _, inst1 := range o.Clusters[cluster] {
```

Comment:
You don't need to iterate over any more.
Also you can create `inst` in the first place as pointer and to return the address:
```go
inst := &Instance{
...

return inst
```